runtime/src/message_processor.rs

```diff
@@ -679,6 +680,16 @@ impl MessageProcessor {
         );
         let keyed_accounts =
             Self::create_keyed_accounts(message, instruction, executable_accounts, accounts)?;
+        for account in &keyed_accounts {
+            if solana_sdk::sysvar::instructions::check_id(account.unsigned_key()) {
+                let mut mut_account_ref = account.try_account_ref_mut()?;
```

Comment:
This block might be better as a `sysvar::instructions` helper

Reply:
yes